plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2912,6 +2915,16 @@ bool producer_plugin_impl::push_read_only_transaction(const transaction_metadata
          return true;
       }
 
+      // when executing on the main thread, need to switch db mode if not already in read only mode
```

Comment:
To be precise, it is when executing on the main thread and in the write window

Reply:
Updated